addons/report/controllers/main.py

```diff
@@ -23,11 +23,12 @@ class ReportController(Controller):
         '/report/<path:converter>/<reportname>/<docids>',
     ], type='http', auth='user', website=True)
     def report_routes(self, reportname, docids=None, converter=None, **data):
-        report_obj = request.registry['report']
-        cr, uid, context = request.cr, request.uid, request.context
-
+        Report = request.env['report']
+        context = dict(request.env.context)
         if docids:
             docids = [int(i) for i in docids.split(',')]
+        objects = Report._get_report_from_name(reportname)
```

Comment:
change variable name instead of objects give some other name. i think it's ir.actions.report.xml instance.
IrActionReport